src/Microsoft.PowerShell.Commands.Utility/commands/utility/CustomSerialization.cs

```diff
@@ -342,7 +342,7 @@ private bool HandlePrimitiveKnownTypePSObject(object source, string property, in
 
             bool sourceHandled = false;
             PSObject moSource = source as PSObject;
-            if (moSource != null && !moSource.immediateBaseObjectIsEmpty)
+            if (moSource != null && !moSource.ImmediateBaseObjectIsEmpty)
             {
                 // Check if baseObject is primitive known type
                 object baseObject = moSource.ImmediateBaseObject;
```

Comment:
Do we need to double check here? Without knowing a great deal about the classes in play here, it looks like you could skip most of the double-checking things in this file and just go right for `object.baseObject = moSource?.ImmediateBaseObject;`

Reply:
I agree. Much more elegant. I haven't focused anything on style but tried to keep the changes as small as possible to make it easier to review. Maybe a separate PR?

Reply:
Fill free to postpone if you have little time now.

Reply:
@powercode if you'd rather not worry about it, just ping me when this gets merged and I'll do a style PR as a follow up. 🙂